llmfoundry/utils/exceptions.py

```diff
@@ -116,6 +116,21 @@ def __init__(self, repeated_role: str) -> None:
         super().__init__(message)
 
 
+class ChatTemplateError(ValueError, ContextualError):
+    """Error thrown when a chat template fails to process a sample."""
+
+    def __init__(
+        self,
+        template: str,
+        sample: List[Dict[str, Any]],
+        inner_message: str,
+    ) -> None:
+        self.template = template
+        self.sample = sample
+        message = f'Failed to process sample {sample} with template {template}. {inner_message}'
```

Comment:
I think putting the template itself into the error is maybe a bit too much text?

Reply:
Our template and the huggingface one are about one-liners. I was thinking including the template could make the error message more actionable. For example the Mistral error says something about repeated roles when the actual issue is no system prompts being allowed, which is fixable by including the mistral template.

Reply:
hmm ok, all good